### PR TITLE
Create html report

### LINKED ISSUE #6: HTML report output for test results
As a user, I want to be able to read the results of the test in HTML.

Acceptance criteria:
- Formatting
- Page for results
- Page for details of the run

### PR DESCRIPTION
**Pull request by andrsbtrg**
### Reference to related issue

Closes: Issue #6 

### What was added/changed?

- Added public method in maple `create_report(output: str)` which creates the report html page at the desired output directory
  
### Why was it added/changed?

- With this function we make it optional to create a report in html, which can be later used in Maple-Ci 

### Technical implementation

- Added `jinja2` as dependency to render some html template
- Created a templates folder inside of maple to contain the html report template

### Acceptance criteria

- [x] Using the `mp.create_report('/home/')` should create a report in html format at /home/ directory.
- [x] The created report should contain a table and a summary of the test specs

### Checklist before requesting review

- [x] Documentation was expanded
- [x] Acceptance criteria are met
- [x] The function was tested by unit tests

### Checklist for reviewers

- [ ] Code checked
- [ ] Acceptance criteria are met

### CODE REVIEW

**Comment by andrsbtrg** [comment]
[maple_report_2024-07-06_22-08-37.html.pdf](https://github.com/user-attachments/files/16117227/maple_report_2024-07-06_22-08-37.html.pdf)

attaching report as pdf because html is not a supported type in a comment

File: src/maple/__init__.py [hunk 1]
```diff
@@ -369,3 +372,16 @@ def print_info(specs):
...
+
+def generate_report(output_path: str):
+    """
+    Generats a report file with the test cases after
+    mp.run()
+
+    Args:
+        output_path: directory to save reports
+    """
+    results = get_test_cases()
```
Comment (Gizemdem) [inline src/maple/__init__.py:385]: throw error if there are no previous tests run results

File: examples/specs.py [hunk 2]
```diff
@@ -1,71 +1,68 @@
+import setup  # noqa
 import maple as mp
 
 
 def main():
     stream_id = "24fa0ed1c3"
     mp.stream(stream_id)
+    mp.set_logging(True)
     mp.run(spec_a, spec_b, spec_c, spec_d, spec_e, spec_f, spec_g)
```
Comment (Gizemdem) [inline examples/specs.py:9]: Improvement: generate report can be called through mp run as an optional argument

File: examples/specs.py [hunk 3]
```diff
@@ -1,71 +1,68 @@
...
+    mp.generate_report(".")
```
Comment (Gizemdem) [inline examples/specs.py:10]: Improvement: generate report can be called through mp run as an optional argument

**Review by Gizemdem — APPROVED** [review]

File: examples/specs.py (hunk 3 above)
Comment (andrsbtrg) [reply examples/specs.py:10]: done :)

**Pull request merged**